src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -0,0 +1,951 @@
+                {
+                    localDeclaration = localDec;
+                }
+            }
+
+            if (localDeclaration is null)
+            {
+                varDeclName = null;
+                return false;
+            }
+
+            // TODO: handle in the future
+            if (syntaxFacts.GetVariablesOfLocalDeclarationStatement(localDeclaration).Count > 1)
+            {
+                varDeclName = null;
+                localDeclaration = null;
+                return false;
+            }
+
+            if (syntaxFacts.IsVariableDeclarator(expressionDecl))
+            {
+                varDeclName = syntaxFacts.GetIdentifierOfVariableDeclarator(expressionDecl).ValueText;
+                return true;
+            }
+
+            varDeclName = null;
+            localDeclaration = null;
+            return false;
+        }
+
+        /// <summary>
+        /// Locates all the call sites of the method that introduced the parameter
+        /// </summary>
+        protected static async Task<ImmutableDictionary<Document, List<SyntaxNode>>> FindCallSitesAsync(
+            Document document, IMethodSymbol methodSymbol, CancellationToken cancellationToken)
+        {
+            var methodCallSitesBuilder = ImmutableDictionary.CreateBuilder<Document, List<SyntaxNode>>();
+            var progress = new StreamingProgressCollector();
+            await SymbolFinder.FindReferencesAsync(
+                methodSymbol, document.Project.Solution, progress,
+                documents: null, FindReferencesSearchOptions.Default, cancellationToken).ConfigureAwait(false);
+            var referencedSymbols = progress.GetReferencedSymbols();
+
+            // Ordering by descending to sort invocations by starting span to account for nested invocations
+            var referencedLocations = referencedSymbols.SelectMany(referencedSymbol => referencedSymbol.Locations)
+                .Distinct().Where(reference => !reference.IsImplicit)
+                .OrderByDescending(reference => reference.Location.SourceSpan.Start);
+
+            // Adding the original document to ensure that it will be seen again when processing the call sites
+            // in order to update the original expression and containing method.
+            methodCallSitesBuilder.Add(document, new List<SyntaxNode>());
+
+            foreach (var refLocation in referencedLocations)
+            {
+                // Does not support cross-language references currently
+                if (refLocation.Document.Project.Language == document.Project.Language)
+                {
+                    var reference = refLocation.Location.FindNode(cancellationToken).GetRequiredParent();
+
+                    // Only adding items that are of type InvocationExpressionSyntax
+                    SyntaxNode invocation;
+                    if (methodSymbol.MethodKind is MethodKind.Constructor && reference is TObjectCreationExpressionSyntax objectCreation)
```

Comment:
i don't think you need the MethodKind check.  you could just remove it.

Reply:
removed